pandas/tests/extension/test_arrow.py

```diff
@@ -2368,3 +2368,12 @@ def test_pickle_old_arrowextensionarray():
     tm.assert_extension_array_equal(result, expected)
     assert result._pa_array == pa.chunked_array(data)
     assert not hasattr(result, "_data")
+
+
+def test_setitem_boolean_replace_with_mask_segfault():
+    # GH#52059
+    N = 145_000
+    arr = ArrowExtensionArray(pa.chunked_array([np.ones((N,), dtype=np.bool_)]))
```

Comment:
I don't think you need this > 145_000. Based on @lukemanley's report, this just happens for any (also tiny) chunked array. 
(I suppose the larger number came from reading where depending on the size it was creating a chunked array or not?)

Reply:
The test did not fail with a smaller size, this was confusing to me as well but could this only get to fail with arrow functionality only

Reply:
OK, that might be a different issue. But so this test is currently not actually testing the replace_with_mask issue giving invalid return arrays, because it seems the `==` is not actually doing its work ... (another bug!)

So also with a N of 2, I see the invalid output, the `==` just passes:
```
In [14]: N = 2

In [15]: arr = pd.arrays.ArrowExtensionArray(pa.chunked_array([np.ones((N,), dtype=np.bool_)]))

In [16]: expected = arr.copy()

In [17]: arr[np.zeros((N,), dtype=np.bool_)] = False

In [18]: expected._data
Out[18]: 
<pyarrow.lib.ChunkedArray object at 0x7fa0f51e2de0>
[
  [
    true,
    true
  ]
]

In [19]: arr._data
Out[19]: 
<pyarrow.lib.ChunkedArray object at 0x7fa0f52e6390>
[
<Invalid array: Buffer #1 too small in array of type bool and length 2: expected at least 1 byte(s), got 0
/home/joris/scipy/repos/arrow/cpp/src/arrow/array/validate.cc:116  ValidateLayout(*data.type)>
]

In [20]: expected._data == arr._data
Out[20]: True
```

Reply:
Using 145_000 causes a segfault on my machine without the change though, so this should be good enough as a test for now?